cmd/limactl/shell.go

```diff
@@ -224,7 +224,7 @@ func shellAction(cmd *cobra.Command, args []string) error {
 	logLevel := "ERROR"
 	// For versions older than OpenSSH 8.9p, LogLevel=QUIET was needed to
 	// avoid the "Shared connection to 127.0.0.1 closed." message with -t.
-	olderSSH := sshutil.DetectOpenSSHVersion(arg0).LessThan(*semver.New("8.9.0"))
+	olderSSH := sshutil.DetectOpenSSHInfo(arg0).Version.LessThan(*semver.New("8.9.0"))
```

Comment:
We can introduce a new exported function `DetectOpenSSHVersion` and use it here and in `copy.go`:

```suggestion
	olderSSH := sshutil.DetectOpenSSHVersion(arg0).LessThan(*semver.New("8.9.0"))
```

Both `DetectOpenSSHInfo` and `OpenSSHInfo` can be unexported.

```go
func DetectOpenSSHInfoVersion(ssh string) semver.Version {
	return detectOpenSSHInfo(ssh).Version
}

func detectOpenSSHInfo(ssh string) openSSHInfo {
	//...
}

// ...
type openSSHInfo struct {
	// Version is set to the version of OpenSSH, or semver.New("0.0.0") if the version cannot be determined.
	Version semver.Version

	// Some distributions omit this feature by default, for example, Alpine, NixOS.
	GSSAPISupported bool
}
```

Reply:
Thanks. With this change, I won't need to modify shell.go and copy.go. :pray: 

https://github.com/lima-vm/lima/compare/4daab365eff3fec89c9f4ff5e7838c6f35cbbc0d..1fbf86db9443418409f31576ed2b1c425a0f736c